src/MarginTrading.Backend.Services/TradingEngine.cs

```diff
@@ -138,6 +141,8 @@ private Task<Order> PlaceMarketOrderByMatchingEngineAsync(Order order, IMatching
                     return false;
                 }
 
+                _volumeEquivalentService.EnrichOpeningOrder(order);
```

Comment:
I feel like enriching the order itself is unnecessary. Is it possible to add this info to the OrderHistory or OrderFullContract?